### DISCUSSION TITLE
Will macOS x86 still be supported for the next few years?

### DISCUSSION THREAD

**Discussion by uwu-420** [jax-ml/jax#22936] [General] [answered]
Pytorch ditched official support for macOS x86 builds, see https://github.com/pytorch/pytorch/issues/114602 and Tensorflow did the same as well, see https://github.com/tensorflow/tensorflow/releases/tag/v2.16.1
Makes me wonder if Jax soon will do the same. I have to choose the deep learning stack for an upcoming project and support for Intel Macs is one of the requirements. Hence the question :)

**Comment by jakevdp** [comment]
Tagging @hawkinsp who has more context on the macOS x86 builds.

**Comment by hawkinsp** [comment, marked as answer]
Hi!

Ultimately we're in the same boat as TF and PyTorch: Mac x86 hardware is becoming increasingly scarce, which makes life harder for engineers to fix the code if it breaks.

We'll make a best effort to keep the Mac x86 builds around and working as long as we can, but they are liable to be removed if:
a) the infrastructure that runs our CI drops support for Mac x86. With no CI, we can't test what we release. We wouldn't intentionally break builds, but with no testing things will break eventually.
b) we hit a problem that we can't easily solve. So far Mac x86 has been reasonably painless to keep working, but if we hit a problem that required significant work we might decide that the time had come to retire that build.

There's no timeline for either. It seems likely that (a) will happen at some point in the future, although I don't know when.

Make sense? We'll do our best, but no guarantees.

**Reply by uwu-420** [reply]
Thanks for the detailed explanation. Makes sense to me. Cheers!

**Comment by uwu-420** [comment]
For people who came here, macos x86_64 support has been removed, see https://jax.readthedocs.io/en/latest/changelog.html#jax-0-5-0-jan-17-2025

**Comment by AnyCPU** [comment]
@uwu-420 following is just some thoughts in the air...

to be honest it looks a little bit suspicious and discriminatious

for instance i have to Intel-based devices with  i3-6006U (Nov 2016) and i7-8559U (Apr 2018),
a first one is an WIndows machine, a second one is MBP

and i can use jax 0.5.* on Windows/WSL2 without any issue at least for quick start.

it is a little bit strange and weird as for me.

it is nice to have a list of steps that might help to support x86 mac as long as possible,
maybe someone (and me) will have a chance to help at spare time.

p.s.: just my 2 cents

**Reply by uwu-420** [reply]
FWIW I switched to the conda ecosystem to continue using up-to-date jax.
See https://anaconda.org/conda-forge/jaxlib and also https://github.com/prefix-dev/pixi/ for a quite delightful next-gen package manager that supports the conda ecosystem but still allows for using PyPI dependencies where necessary.

**Reply by AnyCPU** [reply]
@uwu-420 thanks, it works!
